metaflow/plugins/kubernetes/kubernetes_job.py

```diff
@@ -71,6 +69,21 @@ def create(self):
         # Note: This implementation ensures that there is only one unique Pod
         # (unique UID) per Metaflow task attempt.
         client = self._client.get()
+        tolerations = []
+        for toleration in self._kwargs.get("tolerations") or []:
+            try:
+                tolerations.append(client.V1Toleration(**toleration))
+            except TypeError:
+                raise KubernetesJobException("Toleration definition contains invalid keys: %s" % toleration.keys())
+
+        try:
+            node_selector = {
+                str(k.split("=", 1)[0]): str(k.split("=", 1)[1])
+                for k in self._kwargs.get("node_selector")
+            }
+        except IndexError:
+            raise KubernetesJobException(f"Unable to parse node_selector {self._kwargs.get('node_selector')}")
```

Comment:
@savingoyal I moved `node_selector` as well because now `KUBERNETES_NODE_SELECTOR` is set in `kubernetes_job.py:97`